### DISCUSSION TITLE
exploring-the-dotnet-8-preview-updates-to-docker-images-in-dotnet-8/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#394] [Announcements]
# exploring-the-dotnet-8-preview-updates-to-docker-images-in-dotnet-8/

In this post I describe the changes to .NET 8 docker files in .NET 8 including changes to images, new image types and changes to image tagging

https://andrewlock.net/exploring-the-dotnet-8-preview-updates-to-docker-images-in-dotnet-8/

**Comment by mfe-** [comment]
Thanks for sharing! Which docker images got by default the lib `libgssapi_krb5.so.2 ` ?

**Reply by andrewlock** [reply]
Sorry, I don't know what that is 😅

**Comment by kwaazaar** [comment]
You should mention that those chisled base images do not support running code that uses (EFCore with) **SqlClient**, as SqlClient does not support invariant globalization: https://github.com/dotnet/SqlClient/issues/2239

You will get this error:
`Process terminated. Couldn't find a valid ICU package installed on the system. Please install libicu (or icu-libs)`

**Reply by andrewlock** [reply]
If you use the `jammy-chiseled-extra` images then you don't need to use invariant globalization 🙂

**Comment by josephcorbett** [comment]
Using ASPNETCORE_HTTP_PORTS saved us so much work. Thank you for this information! You are doing the Lord's work.

**Comment by monkeydeus** [comment]
I had quite the kludgey journey trying to get a simple .net8 web api to run in a container today, just because of the localhost issue, but I had to take a rather circuitous route to get to a place where I could understand this post and your post about the loopback issue using localhost. Thanks for putting this out there and clarifying it.  Will be spending more time in the near future reading your other posts on .net 8 and containerization 👍

**Comment by sohlae** [comment]
Hi Andrew, thank you for this article. I was able to fix the issues that I had with my .NET 8 Web API projects. 

I just have a question regarding .NET 8.0 Background Services. I am using a Dapr sidecar with my background service and for some reason when I deployed them in AKS, the sidecar couldn't find the port 8080 even though I have exposed it in the Dockerfile. 

The content of my background service Dockerfile is the same as my web API Dockerfile, but only the web API works. Can you guide me on how to investigate this further?

**Reply by andrewlock** [reply]
If it's solely a "background service" then presumably you're _not_ exposing any ports in your application 🤔  Are you actually running a web app in the service? 

> even though I have exposed it in the Dockerfile

FYI, the `EXPOSE` command in a dockerfile doesn't really _do_ anything, it's purely informational 😅
